src/core/mount.c

```diff
@@ -1074,23 +1076,36 @@ static void mount_enter_unmounting(Mount *m) {
 static void mount_enter_mounting(Mount *m) {
         int r;
         MountParameters *p;
+        bool source_is_dir = true;
 
         assert(m);
 
         r = unit_fail_if_noncanonical(UNIT(m), m->where);
         if (r < 0)
                 goto fail;
 
-        (void) mkdir_p_label(m->where, m->directory_mode);
+        p = get_mount_parameters_fragment(m);
+        if (p && mount_is_bind(p)) {
+                r = is_dir(p->what, /* follow = */ true);
+                if (r < 0 && r != -ENOENT)
+                        log_unit_info_errno(UNIT(m), r, "Failed to determine type of bind mount source '%s', ignoring: %m", p->what);
+                else if (r == 0)
+                        source_is_dir = false;
```

Comment:
```suggestion
                source_is_dir = r > 0;
```

And you can get rid of the initialization :)

Reply:
Then it would be unititialized if `p && mount_is_bind(p)` is `false`.

Reply:
Oops, sorry.